pinecone/client.go

```diff
@@ -498,27 +498,15 @@ func (req CreatePodIndexRequest) TotalCount() *int {
 //		       fmt.Printf("Successfully created pod index: %s", idx.Name)
 //	    }
 func (c *Client) CreatePodIndex(ctx context.Context, in *CreatePodIndexRequest) (*Index, error) {
-	metric := control.IndexMetric(in.Metric)
+	metric := control.CreateIndexRequestMetric(in.Metric)
 	req := control.CreateIndexRequest{
 		Name:      in.Name,
 		Dimension: in.Dimension,
 		Metric:    &metric,
 	}
 
-	//add the spec to req.
-	//because this is defined as an anon struct in the generated code, it must match exactly here.
-	req.Spec = control.CreateIndexRequest_Spec{
-		Pod: &struct {
-			Environment    string `json:"environment"`
-			MetadataConfig *struct {
-				Indexed *[]string `json:"indexed,omitempty"`
-			} `json:"metadata_config,omitempty"`
-			PodType          control.PodSpecPodType   `json:"pod_type"`
-			Pods             *int                     `json:"pods,omitempty"`
-			Replicas         *control.PodSpecReplicas `json:"replicas,omitempty"`
-			Shards           *control.PodSpecShards   `json:"shards,omitempty"`
-			SourceCollection *string                  `json:"source_collection,omitempty"`
-		}{
+	req.Spec = control.IndexSpec{
```

Comment:
Generated anonymous type cleaned up a bit, which is nice.

Reply:
This is nice. The previous impl was unsightly.